File: multicore-f/L1/src/boundary_cell.f
!C
!C***
!C*** BOUNDARY_CELL
!C***
!C
      subroutine BOUNDARY_CELL
      use STRUCT

      implicit REAL*8 (A-H,O-Z)

!C
!C +------+
!C | Zmax |
!C +------+
!C=== 
      IFACTOT= NX * NY      
      ZmaxCELtot= IFACTOT
 
      allocate (ZmaxCEL(ZmaxCELtot))

      icou= 0      
      k   = NZ
      do j= 1, NY
      do i= 1, NX
        icel= (k-1)*IFACTOT + (j-1)*NX + i
        icou= icou + 1
        ZmaxCEL(icou)= icel
      enddo       
      enddo       
!C===
      return
      end

File: L3/cloned/boundary_cell.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>

#include "struct_ext.h"
#include "boundary_cell.h"
#include "allocate.h"

extern int
BOUNDARY_CELL(void)
{
	int IFACTOT;
	int icou, icel, i, j, k;

	IFACTOT = NX * NY;
	ZmaxCELtot = IFACTOT;

	ZmaxCEL = (int *)allocate_vector(sizeof(int),ZmaxCELtot);

	icou = 0;
	k    = NZ - 1;
	for(j=0; j<NY; j++) {
		for(i=0; i<NX; i++) {
			icel = k * IFACTOT + j * NX + i + 1;
			ZmaxCEL[icou] = icel;
			icou++;
		}
	}

	return 0;
}
